lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -205,7 +206,7 @@ def store_content(resource)
       Puppet.info "Content for '#{resource[:source]}' already exists"
     else
       Puppet.info "Storing content for source '#{resource[:source]}'"
-      content = Puppet::FileServing::Content.indirection.find(resource[:source])
+      content = Puppet::FileServing::Content.indirection.find(resource[:source], {:environment => request.environment})
```

Comment:
What happens if the file content cannot be found? I assume it returns nil. Are there tests for this?